### PR TITLE
Testing pre-main-reconciler-validate-go-mod job

### PR DESCRIPTION
**Pull request by TeodorSAP**
/retest

### CODE REVIEW

File: vpath/pjester.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+pjNames:
```
Comment (Halamix2) [inline vpath/pjester.yaml:1]: Pjtester won't work, this file has wrong name

File: templates/data/incubator-buildpack-data.yaml [hunk 2]
```diff
@@ -777,8 +777,26 @@ templates:
...
+                    - jobConfig_default
+                    - jobConfig_presubmit
+                    - extra_refs_test-infra
+                    - image_bootstrap
+                  local:
+                    - reconciler_main_jobConfig
+              - jobConfig:
+                  name: "pre-main-reconciler-validate-go-mod"
+                  # following regexp won't start build if only Markdown files were changed
+                  run_if_changed: '^go\.mod$'
```
Comment (ruanxin) [inline templates/data/incubator-buildpack-data.yaml:794]:
```suggestion
                  run_if_changed: '^go\.mod$'
                  optional: true
```

Comment (ruanxin) [reply templates/data/incubator-buildpack-data.yaml:794]: let's make this job not required

File: prow/jobs/incubator/reconciler/reconciler.yaml [hunk 3]
```diff
@@ -709,6 +709,51 @@ presubmits: # runs on PRs
...
+      branches:
+        - ^main$
+      extra_refs:
+        - org: kyma-project
+          repo: test-infra
+          path_alias: github.com/kyma-project/test-infra
+          base_ref: main
+      spec:
+        containers:
+          - image: "eu.gcr.io/kyma-project/test-infra/bootstrap:v20220427-9543160d"
```
Comment (Ressetkk) [inline prow/jobs/incubator/reconciler/reconciler.yaml:734]: Have you thought about providing your own image with properly configured tools for this job with required packages?

Then you could just use in PodSpec:
```yaml
command: make
args:
  - validate-go-mod
```
without writing bash commands

Comment (ruanxin) [reply prow/jobs/incubator/reconciler/reconciler.yaml:735]: Hi @Ressetkk , thanks for this solution, do we have existing process to publish own image? If the workload is low, we can go this approach, but if our team need to setup and maintain the image release, then it's overkill for this task, this job is just simply want to run a python script, and bootstrap precompiled python, that's basically the reason choose it.

Comment (Ressetkk) [reply prow/jobs/incubator/reconciler/reconciler.yaml:735]: @ruanxin We have no specific process to publish images and we have no strict rules for that on Prow for now. It's up to you on how you provide the image. You can push the image manually to the registry and update it here as you wish, and use the tag here.

**Comment by TeodorSAP** [comment]
PR for test purposes only, and it got too verbose.
Opening a new one...

**Pull request closed without merging**